src/pages/login/Login.tsx

```diff
@@ -1,12 +1,48 @@
 import classNames from 'classnames/bind';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
 
+import { kakaoLoginCallback } from '@/shared/apis/auth/kakaoLogin';
 import Icon from '@/shared/components/Icon';
+import { TokenKeys } from '@/shared/configs/axios';
+import {
+  NEXT_PUBLIC_KAKAO_BASE_URI,
+  NEXT_PUBLIC_KAKAO_CLIENT_ID,
+  NEXT_PUBLIC_KAKAO_REDIRECT_URI,
+  NEXT_PUBLIC_SCOPE,
+} from '@/shared/constants';
+import { getCookie, setCookie } from '@/shared/utils/cookie';
 
 import styles from './Login.module.scss';
 
 const cx = classNames.bind(styles);
 
 const Login = () => {
+  const router = useRouter();
+  const { code } = router.query;
```

Comment:
하단 useEffect가 아닌 쿼리 값을 가져올때 타입을 지정해주면 좋을 것 같아요
```suggestion
  const { code } = router.query as { code: string };
```